cmake/modules/kconfig.cmake

```diff
@@ -146,13 +147,21 @@ else()
   set(_local_TOOLCHAIN_HAS_PICOLIBC n)
 endif()
 
+# Extract git commit information for application
+git_commit_hash(${APPLICATION_SOURCE_DIR} GIT_COMMIT_HASH)
+if(GIT_COMMIT_HASH)
```

Comment:
this then falls back into the trap, if `APPLICATION_SOURCE_DIR` is not a git directory and `GIT_COMMIT_HASH` is previously set, the value will remain here and will wrongly be treated as the application git commit hash